torch/_decomp/decompositions.py

```diff
@@ -1042,6 +1042,16 @@ def logit_backward(
         )
 
 
+@register_decomposition(aten.dropout)
+@aten.dropout.default.py_impl(DispatchKey.CompositeImplicitAutograd)
+@aten.dropout.default.py_impl(DispatchKey.Autograd)
+def dropout(input: Tensor, p: float, train: Optional[bool]):
+    if train and p != 0:
+        return aten.native_dropout(input, p, train)[0]
+    else:
+        return input
```

Comment:
I hope this is what the real op does (return the input directly), because this can cause some gnarly bugs if it isn't (especially because you are registering a regular decomp for this op)

Reply:
@bdhirsh I remember you mentioned that `native_dropout` returns a clone in both train and eval branches. Is this OK?

Reply:
both native_dropout_cpu and native_droput_cuda returns a 
```
    output = input.clone();
```

IIUC, this is different from returning the input direction. This is a difference instance of the tensor with different storage. 

I need to update this to `return input.clone()`